src/Socket/messages-recv.ts

```diff
@@ -325,6 +325,59 @@ export const makeMessagesRecvSocket = (config: SocketConfig) => {
 		}
 	}
 
+	const handleNewsletterNotification = (id: string, node: BinaryNode) => {
+        const messages = getBinaryNodeChild(node, 'messages')
+        const message = getBinaryNodeChild(messages, 'message')!
+
+        const server_id = message.attrs.server_id
+
+        const reactionsList = getBinaryNodeChild(message, 'reactions')
+		const viewsList = getBinaryNodeChildren(message, 'views_count')
+
+        if(reactionsList){
+			const reactions = getBinaryNodeChildren(reactionsList, 'reaction')
+			if(reactions.length === 0){
+				ev.emit('newsletter.reaction', {id, server_id, reaction: { removed: true }})
+			}
+			reactions.forEach(item => {
+				ev.emit('newsletter.reaction', {id, server_id, reaction: { code: item.attrs?.code, count: +item.attrs?.count }})
+			})
+        }
+
+        if(viewsList.length){
+			viewsList.forEach(item => {
+            	ev.emit('newsletter.view', {id, server_id, count: +item.attrs.count})
+			})
+        }
+	}
+
+	const handleMexNewsletterNotification = (id: string, node: BinaryNode) => {
+		const operation = node?.attrs.op_name
+		const content = JSON.parse(node?.content?.toString()!)
+
+		let contentPath
+
+		if(operation === MexOperations.PROMOTE || operation === MexOperations.DEMOTE){
```

Comment:
need to simplify the logic

```ts
if(operation === MexOperations.PROMOTE){
	action = 'promote'
	contentPath = content.data[XWAPaths.PROMOTE]
}

if(operation === MexOperations.DEMOTE){
	action = 'demote'
	contentPath = content.data[XWAPaths.DEMOTE]
}

if(operation === MexOperations.UPDATE){
	contentPath = content.data[XWAPaths.METADATA_UPDATE]
	ev.emit('newsletter-settings.update', {id, update: contentPath.thread_metadata.settings as NewsletterSettingsUpdate})
}
```

Reply:
i think is better to use switch case, what do you think?

Reply:
yes use a switch